packages/block-editor/src/components/block-list/block.native.scss

```diff
@@ -134,6 +59,28 @@
 }
 
 .neutralToolbar {
-	margin-left: -$block-edge-to-content;
-	margin-right: -$block-edge-to-content;
+	margin-left: - 14;
+	margin-right: - 14;
+}
+
+.solidBorder {
+	position: absolute;
+	top: -12;
+	bottom: 0;
+	left: -12;
+	right: -12;
+	border-width: $block-selected-border-width;
+	border-radius: 4px;
+	border-style: solid;
+}
+
+.dashedBorder {
+	position: absolute;
+	top: -6;
+	bottom: -6;
+	left: -6;
+	right: -6;
+	border-width: $block-selected-border-width;
+	border-radius: 2px;
+	border-style: dashed;
 }
```

Comment:
As discussed on call we can consider move numeric values ( -6,-12 etc) to `_variables.scss` to put some constrain on border layout and make sure each time we apply changes it will be spread to all necessary place.